lib/node_modules/@stdlib/string/len/lib/should-break.js

```diff
@@ -0,0 +1,165 @@
+var ExtendedPictographic = require( './constants.js' ).Extended_Pictographic;
+
+
+// MAIN //
+
+/**
+* Returns whether a break is allowed between the two given grapheme breaking classes according to UAX #29 3.1.1 Grapheme Cluster Boundary Rules on extended grapheme clusters.
+*
+* @param {number} start - start
+* @param {Array<number>} mid - mid
+* @param {number} end - end
+* @param {number} startEmoji - startEmoji
+* @param {Array<number>} midEmoji - midEmoji
+* @param {number} endEmoji - endEmoji
+* @returns {number} break type
+*/
+function shouldBreak( start, mid, end, startEmoji, midEmoji, endEmoji ) {
+	var prevNonExtendIdx;
+	var regIndRange;
+	var nextEmoji;
+	var allEmoji;
+	var previous;
+	var next;
+	var all;
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if extend
+	*/
+	function checker( c ) {
+		return c === RegionalIndicator;
+	}
+	/**
+	* Function expression.
+	* @private
+	* @param {number} c - temp
+	* @returns {boolean} - check if Extend
+	*/
+	function checkerE( c ) {
+		return c === Extend;
+	}
+	all = [ start ].concat( mid ).concat( [ end ] );
+	allEmoji = [ startEmoji ].concat( midEmoji ).concat( [ endEmoji ] );
+	previous = all[ all.length - 2 ];
+	next = end;
+	nextEmoji = endEmoji;
+
+	regIndRange = all.lastIndexOf( RegionalIndicator );
+	if ( regIndRange > 0 &&
+		all.slice( 1, regIndRange ).every( checker ) &&
+		[ Prepend, RegionalIndicator ].indexOf( previous ) === -1 ) {
+		if ( all.filter( checker ).length % 2 === 1 ) {
+			return BreakLastRegional;
+		}
+		return BreakPenultimateRegional;
+	}
+
+	// GB3: CR × LF
+	if ( previous === CR && next === LF ) {
+		return NotBreak;
+	}
+	// GB4: (Control|CR|LF) ÷
+	else if ( previous === Control || previous === CR || previous === LF ) { // eslint-disable-line no-else-return
+		return BreakStart;
+	}
+	// GB5: ÷ (Control|CR|LF)
+	else if ( next === Control || next === CR || next === LF ) {
+		return BreakStart;
+	}
+	// GB6: L × (L|V|LV|LVT)
+	else if ( previous === L &&
+		( next === L || next === V || next === LV || next === LVT ) ) {
+		return NotBreak;
+	}
+	// GB7: (LV|V) × (V|T)
+	else if ( (previous === LV || previous === V ) &&
+		( next === V || next === T ) ) {
+		return NotBreak;
+	}
+	// GB8: (LVT|T) × (T)
+	else if ( ( previous === LVT || previous === T ) &&
+		next === T ) {
+		return NotBreak;
+	}
+	// GB9: × (Extend|ZWJ)
+	else if ( next === Extend || next === ZWJ ) {
+		return NotBreak;
+	}
+	// GB9a: × SpacingMark
+	else if ( next === SpacingMark ) {
+		return NotBreak;
+	}
+	// GB9b: Prepend ×
+	else if ( previous === Prepend ) {
+		return NotBreak;
+	}
+
+	// GB11: \p{Extended_Pictographic} Extend* ZWJ × \p{Extended_Pictographic}
+	prevNonExtendIdx = allEmoji.slice( 0, -1 ).lastIndexOf( ExtendedPictographic ); // eslint-disable-line max-len
+	if ( prevNonExtendIdx !== -1 &&
+		allEmoji[ prevNonExtendIdx ] === ExtendedPictographic &&
+		all.slice( prevNonExtendIdx + 1, -2 ).every( checkerE ) &&
```

Comment:
Similar comment to the above. Instead of a slice (copy), we can have an `every` utility which has an "ndarray" interface which accepts an `offset` parameter specifying the starting index from which to begin searching. See `@stdlib/stats/base/*`.

Reply:
I don't really understand this, how would this utility function be any different from what is currently happening? As far as I can think if we are to use `Array.prototype.every()` in the underlying implementation of this offset function, we will need to create a subset (slice/copy) of the original array.
Or are you suggesting that this function should check for the condition for all the elements in the specified range by iterating over the elements in range and return true/false accordingly?

Reply:
I meant that we'd avoid using prototype functions altogether. For example,

```javascript
function every( N, x, strideX, offsetX, predicate ) {
    var ix;
    var i;

    ix = offset;
    for ( i = 0; i < N; i++ ) {
        if ( predicate( x[ ix ] ) === false ) {
            return false;
        }
        ix += strideX;
    }
    return true;
}
```

The above code would then become

```javascript
N = all.length - 2 - prevNonExtendIdx - 1;
every( N, all, 1, prevNonExtendIdx+1, checkerE );
```

or something similar.

(1) no slice would then be necessary; we'd always iterate over the same view; (2) no prototype functions used; (3) is general. Further, rather than having a callback, the condition which the predicate is wrapping could just be inlined in order to avoid having to create a new function scope for each invocation.

Realistically, we should have a package exposing such a utility, which can be added in a future PR. For now, including such a utility in this implementation would be fine.

In short, we generally avoid relying on built-ins for three reasons (1) general performance, (2) avoiding unnecessary copies, and (3) avoid unnecessary scope creation and destruction.